src/items.rs

```diff
@@ -790,6 +759,37 @@ impl<'a> FmtVisitor<'a> {
     }
 }
 
+fn rewrite_explicit_self(explicit_self: &ast::ExplicitSelf, args: &[ast::Arg]) -> String {
```

Comment:
I guess maybe return a custom enum then? Alternatively we could always return Option<Option<String>> if we implement Rewrite (which would require making Rewrite more generic I guess). If we have to treat the result specially if its an empty string, then it really ought to be explicit by using an enum of some kind.